redbot/launcher.py

```diff
@@ -0,0 +1,383 @@
+    if dev:
+        package = "git+https://github.com/Cog-Creators/Red-DiscordBot@V3/develop"
+        if egg_l:
+            package += "#egg=Red-DiscordBot[{}]".format(", ".join(egg_l))
+    else:
+        package = "Red-DiscordBot"
+        if egg_l:
+            package += "[{}]".format(", ".join(egg_l))
+    code = subprocess.call([
+        interpreter, "-m",
+        "pip", "install", "-U",
+        "--process-dependency-links",
+        package
+    ])
+    if code == 0:
+        print("Red has been updated")
+    else:
+        print("Something went wrong while updating!")
+
+    # If redbot wasn't updated, we renamed our .exe file and didn't replace it
+    scripts = os.listdir(os.path.dirname(launcher_script))
+    if renamed and "redbot-launcher.exe" not in scripts:
+        print("Renaming {} to {}".format(new_name, old_name))
+        os.rename(new_name, old_name)
+
+
+def run_red(selected_instance, autorestart: bool=False, cliflags=None):
+    while True:
+        print("Starting {}...".format(selected_instance))
+        cmd_list = ["redbot", selected_instance]
+        if cliflags:
+            cmd_list += cliflags
+        status = subprocess.call(cmd_list)
+        if (not autorestart) or (autorestart and status != 26):
+            break
+
+
+def cli_flag_getter():
+    print("Would you like to enter any cli flags to pass to redbot? (y/n)")
+    resp = user_choice()
+    if resp == "n":
+        return None
+    elif resp == "y":
+        flags = []
+        print("Ok, we will now walk through choosing cli flags")
+        print("Would you like to specify an owner? (y/n)")
+        choice = user_choice()
+        if choice == "y":
+            print("Enter the user id for the owner")
+            owner_id = user_choice()
+            flags.append("--owner {}".format(owner_id))
+        print("Would you like to specify any co-owners? (y/n)")
+        choice = user_choice()
+        if choice == "y":
+            print("Enter the user ids for the co-owners, separated by a space")
+            co_owner_ids = user_choice().split()
+            for co in co_owner_ids:
+                flags.append("--co-owner {}".format(co))
+        print("Would you like to specify any prefixes? (y/n)")
+        choice = user_choice()
+        if choice == "y":
+            print(
+                "Enter the prefixes, separated by a space (please note "
+                "that this will not handle prefixes containing a space)")
+            prefixes = user_choice().split()
+            for p in prefixes:
+                flags.append("-p {}".format(p))
+        print("Would you like to disable console input? (y/n)")
+        choice = user_choice()
+        if choice == "y":
+            flags.append("--no-prompt")
+        print("Would you like to start with no cogs loaded? (y/n)")
+        choice = user_choice()
+        if choice == "y":
+            flags.append("--no-cogs")
+        print("Is this a selfbot? (y/n)")
```

Comment:
if it's not allowed (but we don't want to shut off the capability), maybe we should make this more hidden as well